aspnetcore/fundamentals/dependency-injection/samples/3.x/DependencyInjectionSample2/Models/Operation.cs

```diff
@@ -0,0 +1,35 @@
+using System;
+using DependencyInjectionSample.Interfaces;
+
+namespace DependencyInjectionSample.Models
+{
+    #region snippet1
+    public class Operation : IOperationTransient, 
+        IOperationScoped, 
+        IOperationSingleton, 
+        IOperationSingletonInstance
+    {
+        public Operation() : this(DateTime.Now.ToString("hh:mm:ss.fff"))
```

Comment:
@serpent5  I changed this in my proposed version to use a time string so you can easily see what's updated when you refresh the screen. Forced to use `DoWork` so we get different times. Impossible for the human eye to compare GUIDs but time is easy.

Reply:
@dougbu  do you approve of me changing from GUID to time string? It makes it easy to see what has changed. The down side is you end up with `DoWork`, but I can explain that.

Reply:
I think I prefer the GUID approach for this because it avoids the `DoWork` problem and I expect developers are used to GUIDs. Here's an example of some formatting changes that _might_ look a bit better with GUIDs:

![image](https://user-images.githubusercontent.com/6025110/88713449-3a3ecf00-d113-11ea-83fe-e10182506851.png)

Reply:
How about using a 3 digit random number and moving the **Operations service** to middleware - not showing this in the view, showing it in the logger.

Reply:
Or I could stick to GUID